site/app/views/grading/ElectronicGraderView.php

```diff
@@ -1358,6 +1364,15 @@ public function renderSubmissionPanel(GradedGradeable $graded_gradeable, int $di
         if ($this->core->getUser()->getGroup() == User::GROUP_STUDENT) {
             $student_grader = true;
         }
+        $instructor = false;
+        $limimted_access_grader = false;
+        $full_access_grader = false;
+        if ($this->core->getUser()->getGroup() == User::GROUP_INSTRUCTOR || $this->core->getUser()->getGroup() == User::GROUP_FULL_ACCESS_GRADER || $this->core->getUser()->getGroup() == User::GROUP_LIMITED_ACCESS_GRADER) {
+            $instructor = true;
+            $limimted_access_grader = true;
+            $full_access_grader = true;
+        }
```

Comment:
You could just check if grading access is allowed for the user which would cleanup the code a lot.